packages/elements/src/collapse/index.ts

```diff
@@ -195,15 +202,17 @@ export class Collapse extends BasicElement {
    */
   protected render (): TemplateResult {
     return html`
-      <ef-header part="header" level="${this.level}" @tap="${this.handleTap}">
-        <ef-icon icon="right" slot="left" part="toggle"></ef-icon>
+      <ef-header part="header" level="${this.level}" @tap=${this.handleTap} role="heading" aria-level=${this.ariaLevel}>
```

Comment:
I think the entire `ef-header` is a clickable element, so it needs `role="button"`. As well, there is no need to set `aria-level` as the semantics is only valid for the light DOM.

This probably needs testing, but I think the entire `ef-collapse` element should have `role=heading`.

Reply:
Your solutions are not valid, I have already tested,

1. Entire `ef-collapse` cannot be a `role=heading` because it makes semantics incorrect, the header and the content must have a separate container place sibling with each other.
2. The `ef-header` cannot be a `role=button` because developers can put a button, link, or icon in the slot of `ef-collapse` in left or right slots which makes semantic meaning and screen reader announce incorrectly

Reply:
You made valid points, however as we have discussed these needs further research.